libkbfs/kbfs_ops_test.go

```diff
@@ -3904,3 +3904,65 @@ func TestKBFSOpsReadonlyFSNodes(t *testing.T) {
 	require.Equal(t, int64(5), n)
 	require.Equal(t, "ddata", string(data))
 }
+
+type mdServerShutdownOverride struct {
+	mdServerLocal
+	override bool
+}
+
+func (md mdServerShutdownOverride) isShutdown() bool {
+	if md.override {
+		return true
+	}
+	return md.mdServerLocal.isShutdown()
+}
+
+func TestKBFSOpsReset(t *testing.T) {
+	var u1 kbname.NormalizedUsername = "u1"
+	config, _, ctx, cancel := kbfsOpsConcurInit(t, u1)
+	defer kbfsConcurTestShutdown(t, config, ctx, cancel)
+	md := &mdServerShutdownOverride{config.MDServer().(mdServerLocal), false}
+	config.SetMDServer(md)
+
+	err := EnableImplicitTeamsForTest(config)
+	require.NoError(t, err)
+
+	t.Log("Create a private folder.")
+	name := "u1"
+	h, err := ParseTlfHandle(
+		ctx, config.KBPKI(), config.MDOps(), string(name), tlf.Private)
+	require.NoError(t, err)
+	kbfsOps := config.KBFSOps()
+	rootNode, _, err := kbfsOps.GetOrCreateRootNode(ctx, h, MasterBranch)
+	require.NoError(t, err)
+
+	oldID := h.tlfID
+	t.Logf("Make a new revision for TLF ID %s", oldID)
+	_, _, err = kbfsOps.CreateDir(ctx, rootNode, "a")
+	require.NoError(t, err)
+	err = kbfsOps.SyncAll(ctx, rootNode.GetFolderBranch())
+	require.NoError(t, err)
+
```

Comment:
I'm a bit confused; There doesn't seem to be a place where we blow away the TLF to simulate the server side reset. How did it work?

Also it'd be nice to test the `FolderNotResetOnServer` case too if it's not too complicated.

Reply:
The local mdserver implementation doesn't store the handle -> ID mapping for implicit team TLFs, so it doesn't have the ability to do the same checks as the real mdserver.  So every `GetForHandle` call returns `ServerErrorClassicTLFDoesNotExist{}`.

If you think it's important I can change the disk-based mdserver to track those mappings, and have the test delete it explicitly.  I'm a little worried it'll break other tests, so I can't say how easy it will be.

(I have tested this manually against the real mdserver, for what it's worth.)

Reply:
Ah got it. As you've already tested against the real mdserver, I'm good with this. Thanks for the context!